src/main/js/wiki/components/BranchDropdown.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import injectSheet from "react-jss";
+import classNames from "classnames";
+import { orderBranches } from "../../repository/components/BranchOverview";
+
+const styles = {
+  dropdown: {
+    "background-color": "white",
+    border: "#777777",
+    color: "#777777",
+    padding: "6px 12px",
+    fontSize: "16px",
+    marginLeft: "auto"
+  }
+};
+
+export type Branch = {
+  name: string;
+  link: string;
+};
+
+type Props = {
+  branches: Branch[];
+  currentBranch: string;
+  classes: any;
+};
+
+export const buildBranchUrl = (oldBranchName: string, newBranchName: string) => {
+  const url = window.location.href;
+  return url.replace(oldBranchName, newBranchName);
```

Comment:
This looks like a bad idea. What happens if we have a branch named `smeagol` for example?

Reply:
Hmm, good Point. Should I try to find a more accurate regex expression here or is there another, better way to switch the branch?

Reply:
replaced with history.push()